File: webui/src/test/scala/com/taskadapter/webui/uiapi/NewConfigSuggesterTest.scala
package com.taskadapter.webui.uiapi

import com.taskadapter.common.ui.{FieldMapping, NewConfigSuggester}
import com.taskadapter.connector.jira.JiraFactory
import com.taskadapter.connector.redmine.RedmineFactory
import com.taskadapter.model.{AllFields, GUser}
import org.junit.runner.RunWith
import org.scalatest.concurrent.ScalaFutures
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class NewConfigSuggesterTest extends FunSpec with ScalaFutures with Matchers {

  val defaultRedmineFields = new RedmineFactory().getDefaultFieldsForNewConfig
  val defaultJiraFields = new JiraFactory().getDefaultFieldsForNewConfig

  val jiraRedmineFieldsNumber = 8

  val noUser : GUser = null.asInstanceOf[GUser]

  it("suggests all elements from left connector") {
    val list = NewConfigSuggester.suggestedFieldMappingsForNewConfig(
      defaultRedmineFields,
      defaultJiraFields)

    list.size shouldBe jiraRedmineFieldsNumber
    list.contains(new FieldMapping(AllFields.assigneeLoginName, AllFields.assigneeLoginName, true, null)) shouldBe true
  }

  it("suggests all elements from right connector") {
    val list = NewConfigSuggester.suggestedFieldMappingsForNewConfig(
      defaultJiraFields,
      defaultRedmineFields)

    list.size shouldBe jiraRedmineFieldsNumber
  }

}

File: webui/src/test/java/com/taskadapter/web/uiapi/NewConfigSuggesterTest.java
package com.taskadapter.web.uiapi;

import com.taskadapter.common.ui.FieldMapping;
import com.taskadapter.common.ui.NewConfigSuggester;
import com.taskadapter.connector.jira.JiraFactory;
import com.taskadapter.connector.redmine.RedmineFactory;
import com.taskadapter.model.AllFields;
import com.taskadapter.model.Field;
import com.taskadapter.model.GUser;
import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class NewConfigSuggesterTest {
    private static final List<Field<?>> defaultRedmineFields = new RedmineFactory().getDefaultFieldsForNewConfig();
    private static final List<Field<?>> defaultJiraFields = new JiraFactory().getDefaultFieldsForNewConfig();

    private static final int jiraRedmineFieldsNumber = 8;

    @Test
    public void suggestsAllElementsFromLeftConnector() {
        var list = NewConfigSuggester.suggestedFieldMappingsForNewConfig(
                defaultRedmineFields,
                defaultJiraFields);

        assertThat(list).hasSize(jiraRedmineFieldsNumber);
        assertThat(list).contains(
                new FieldMapping(AllFields.assigneeLoginName, AllFields.assigneeLoginName, true, null));
    }

    @Test
    public void suggestsAllElementsFromRightConnector() {
        var list = NewConfigSuggester.suggestedFieldMappingsForNewConfig(
                defaultJiraFields,
                defaultRedmineFields);

        assertThat(list).hasSize(jiraRedmineFieldsNumber);
    }
}
